### ISSUE TITLE
Prevent admin from re-rendering on form update

### ISSUE THREAD

**Issue by kyledurand** [Shopify/polaris-react#3575]
We need to prevent action measuring when screen is vertically resized

[Screencast](https://screenshot.click/28-49-28470-94174.webm)

**Comment by AsgardLeonardo** [comment]
Hi guys, 
Found this GH via this [Slack convo](https://shopify.slack.com/archives/CJJD1Q7PT/p1603896479291000) in # design-language.
I have a store reporting the same type of "glitch" but in the collections page:

They have some collections with rather long names which seem to be "out of bounds" and on top of the menus, causing the collection title to flicker and not be able to interact with the menus themselves and actually causing the whole store admin to eventually crash (in Chrome) needing to close the tab and every other Shopify tab.

If the New Design Language is disabled from our view, the (old) collection page shows just fine.

So far I have not been able to replicate this on my test store, not even with a super long name (and under the New Design Language) 

Leaving [another screencast](https://screenshot.click/30-58-bd66s-9gk4q.mp4) :) 

Merchant store [collection example](https://bird-and-earth.myshopify.com/admin/collections/153759055960) 
[Internal](https://app.shopify.com/services/internal/shops/26787905624)
[ZD Ticket](https://shopify.zendesk.com/agent/tickets/20363140)

**Comment by kyledurand** [comment]
Thanks @AsgardLeonardo I'll make this a priority today. cc @dfmcphee looks like we still have an infinite render issue

**Comment by kyledurand** [comment]
Some additional context: https://github.com/Shopify/polaris-react/pull/3576#issuecomment-719727459

**Comment by kyledurand** [comment]
And a bit more context. I found that the re-rendering of page actions on some pages is actually legit. 

On the product details page, [here](https://github.com/Shopify/web/blob/master/app/sections/Products/ProductDetails/ProductDetails.tsx#L1888-L1893), the preview action takes a new action on every change made to the product form. This is a sort of cached preview of anything that's been changed since save.

We'll have to be more explicit about when our actions do the measuring

**Comment by oldcheddar** [comment]
hey team 👋  

I'm just adding an instance for tracking as I see this has already been prioritized 🙌  

* [looks like this](https://screenshot.click/09-34-67c7i-kue3a.mp4) most of the merchant's products have longer titles so this is happening on the majority of their products.

[ZenDesk](https://shopify.zendesk.com/agent/tickets/20347870)
[Internal](https://app.shopify.com/services/internal/shops/5822578748)

**Comment by Rony-C** [comment]
Hi all,

I see this is closed but the issue is still persisting on the merchants store. 

* [Affected product](https://shopurotuning.myshopify.com/admin/products/4598051602487)
* [Video here](https://screenshot.click/m7exb_2020-11-16.mp4)

I'm wondering if this is the same issue? It looks the same as the original video linked posted by Kyle

**Comment by kyledurand** [comment]
Thanks Rony, it was automatically closed when that PR shipped but we're still waiting on the release to hit web. https://github.com/Shopify/web/pull/34757. I'll close this when that's been pushed to production

**Comment by kyledurand** [comment]
This fix is in production now, feel free to re-open if the bug persists

**Comment by PaddyMoloney** [comment]
Hey folks, looks like this is still happening on certain stores https://app.shopify.com/services/internal/shops/10195793

**Comment by kyledurand** [comment]
Thanks Paddy, we had to revert the fix unfortunately. We have another PR up but it won't be shipped til after BFCM

**Comment by TaylorCyr** [comment]
Hi Kyle. Before the fix was reverted, I also had a merchant mention that the fix left his product title rendering like this:

![](https://screenshot.click/2020-11-2417-43_ayoj3-8ww2j_667_24r.jpg)

At the time I was able to replicate, on Chrome. This product:

https://oceanside-photo-telescope.myshopify.com/admin/products/11230504202

**Comment by mediumjones** [comment]
Hey @kyledurand not sure if this had anything to do with it, but we updated the page title recently to add product status badges to it. I can try to dig up the PR if you think it'll help. 

https://github.com/Shopify/web/issues/30082

**Comment by patricksayegh** [comment]
Hey team! I have a merchant who is having this issue(I am also able to replicate). From what I found(awaiting the merchant's confirmation as well) it only appears to be happening on _one_ of their products. If you duplicate that product in question, the issue no longer persists on the duplicate - which a workaround could be hiding the affected product and have the merchant use the duplicate. Details are as followed:

[Internal](https://app.shopify.com/services/internal/shops/639991855)
[Product in question](https://sir-jacks.myshopify.com/admin/products/4634128777285)
[Duplicate of that product, that works](https://sir-jacks.myshopify.com/admin/products/4647179452485)
[Video of issue](https://screenshot.click/03-17-59367-95532.webm)
[Zendesk ticket](https://shopify.zendesk.com/agent/tickets/21597078)
[Incident Dashboard](https://incidents.shopify.io/incidents/175)

I haven't asked the merchant to delete the product in question, but to hide it and use the duplicated one. I will add more products if they have more and if you require so. 

Thanks so much!

**Comment by TaylorCyr** [comment]
Hey @kyledurand, looks like the PR to address this has been in review for a while. Now that its post-holidays, any chance that could get pushed through please?

**Comment by Fox04k** [comment]
Hi Guys, just adding another example here. I could not replicate this on my side but merchant has sent a vid of it happening to them. 

[Product](https://soft-and-cozy-rugs.myshopify.com/admin/products/4842191552591)
[Internal](https://app.shopify.com/services/internal/shops/27865448527)
[ZD Ticket](https://shopify.zendesk.com/agent/tickets/21621066)
[Video](https://www.icloud.com/attachment/?u=https%3A%2F%2Fcvws.icloud-content.com%2FB%2FAaDOhICM209UudsHTDsZIMoATao6AZfFijWsp2nujN7BPL8mEYvti9bY%2F%24%7Bf%7D%3Fo%3DAojuxFKfwkThGVAwwFs6z_hvAhK2jowmWrKyR5-W_yxn%26v%3D1%26x%3D3%26a%3DCAogma0nxCHk0OODxCTSO72hXQFbW_dwJgiGBRqMcwJSOIwSeBD95NiE7S4Y_fTT2PYuIgEAKgkC6AMA_2lkGRxSBABNqjpaBO2L1thqJmwtFukmQ-f2_RTfYxn28scxPcDk-U5tT1zgzE_U9BZhRvNJrVAtciazfd_mZmFeaBmu5zCzcCwRhOX_Ib1_aQhRgnoj67wO-s8Ska-x6w%26e%3D1612409272%26fl%3D%26r%3D7D497954-A4D5-43FD-B9E9-508CB34909A3-1%26k%3D%24%7Buk%7D%26ckc%3Dcom.apple.largeattachment%26ckz%3D402CF739-5452-44A7-8AE8-EE0D421274D5%26p%3D34%2)

Seems to only be happening on one of their products, awaiting confirmation of this.

**Comment by kyledurand** [comment]
Thanks folks, we understand the frustration behind this but also want to be careful we don't introduce any other bugs. Hoping to roll out a fix early next week

**Comment by Drake2300** [comment]
@kyledurand Just double checking that closing with that commit was intentional, and this should now be resolved?

**Comment by kyledurand** [comment]
Whoops, thanks Drake. This will be fixed Monday when we ship the Polaris update to web

**Comment by kyledurand** [comment]
This shipped yesterday and seems to be stable. Merchants affected should only, in the worst case scenario, see one shift of title content when making updates to product pages

i.e https://screenshot.click/screencast_2021-01-14_13-49-52.mp4

This is because the title and secondary actions are competing for space and on a re-render the JS is somewhat competing with css in determining what can / should be rendered. There's no elegant solution to this as this component needs to support long titles with a few secondary actions, or short titles with lots of secondary actions and we need to maximize what gets shown here.

Feel free to reopen if anything else gets weird
